clang/lib/CIR/Dialect/Transforms/TargetLowering/LowerFunction.cpp

```diff
@@ -32,6 +32,210 @@ using ABIArgInfo = ::cir::ABIArgInfo;
+    llvm_unreachable("NYI");
+  }
+
+  llvm::TypeSize DstSize = CGF.LM.getDataLayout().getTypeAllocSize(DstTy);
+
+  // If store is legal, just bitcast the src pointer.
+  assert(!::cir::MissingFeatures::vectorType());
+  if (SrcSize.getFixedValue() <= DstSize.getFixedValue()) {
+    // Dst = Dst.withElementType(SrcTy);
+    CGF.buildAggregateStore(Src, Dst, DstIsVolatile);
+  } else {
+    llvm_unreachable("NYI");
+  }
+}
+
+// FIXME(cir): Create a custom rewriter class to abstract this away.
+Value createBitcast(Value Src, Type Ty, LowerFunction &LF) {
+  return LF.getRewriter().create<CastOp>(Src.getLoc(), Ty, CastKind::bitcast,
+                                         Src);
+}
+
+/// Coerces a \param Src value to a value of type \param Ty.
+///
+/// This safely handles the case when the src type is smaller than the
+/// destination type; in this situation the values of bits which not present in
+/// the src are undefined.
+///
+/// NOTE(cir): This method has partial parity with CGCall's CreateCoercedLoad.
+/// Unlike the original codegen, this function does not emit a coerced load
+/// since CIR's type checker wouldn't allow it. Instead, it casts the existing
+/// ABI-agnostic value to it's ABI-aware counterpart. Nevertheless, we should
+/// try to follow the same logic as the original codegen for correctness.
+Value createCoercedValue(Value Src, Type Ty, LowerFunction &CGF) {
+  Type SrcTy = Src.getType();
+
+  // If SrcTy and Ty are the same, just reuse the exising load.
+  if (SrcTy == Ty)
+    return Src;
+
+  // If it is the special boolean case, simply bitcast it.
+  if ((isa<BoolType>(SrcTy) && isa<IntType>(Ty)) ||
+      (isa<IntType>(SrcTy) && isa<BoolType>(Ty)))
+    return createBitcast(Src, Ty, CGF);
+
+  llvm::TypeSize DstSize = CGF.LM.getDataLayout().getTypeAllocSize(Ty);
+
+  if (auto SrcSTy = dyn_cast<StructType>(SrcTy)) {
+    Src = enterStructPointerForCoercedAccess(Src, SrcSTy,
+                                             DstSize.getFixedValue(), CGF);
+    SrcTy = Src.getType();
+  }
+
+  llvm::TypeSize SrcSize = CGF.LM.getDataLayout().getTypeAllocSize(SrcTy);
+
+  // If the source and destination are integer or pointer types, just do an
+  // extension or truncation to the desired type.
+  if ((isa<IntType>(Ty) || isa<PointerType>(Ty)) &&
+      (isa<IntType>(SrcTy) || isa<PointerType>(SrcTy))) {
+    llvm_unreachable("NYI");
+  }
+
+  // If load is legal, just bitcast the src pointer.
+  if (!SrcSize.isScalable() && !DstSize.isScalable() &&
+      SrcSize.getFixedValue() >= DstSize.getFixedValue()) {
+    // Generally SrcSize is never greater than DstSize, since this means we are
+    // losing bits. However, this can happen in cases where the structure has
+    // additional padding, for example due to a user specified alignment.
+    //
+    // FIXME: Assert that we aren't truncating non-padding bits when have access
+    // to that information.
+    // Src = Src.withElementType();
```

Comment:
Is this FIXME copy and paste? How is someone suppose to find the problem they might have is because of this?